builder/build_env.py

```diff
@@ -234,19 +236,66 @@ def _create_recipes(repository, recipes, variant_config_files, variants, channel
     for recipe in config_data.get('recipes', []):
         if recipes and not recipe.get('name') in recipes:
             continue
-        packages, deps = _get_package_dependencies(recipe.get('path'),
-                                                   variant_config_files,
-                                                   variants)
+        packages, deps, versioned_deps = _get_package_dependencies(recipe.get('path'),
+                                                                   variant_config_files,
+                                                                   variants)
         output = { 'recipe' : recipe.get('name', None),
                    'repository' : repository,
                    'packages' : packages,
                    'dependencies' : deps,
+                   'versioned_dependencies' : versioned_deps,
                    'channels' : channels if channels else []}
         outputs.append(output)
```

Comment:
@bnemanich
You can see here I added versioned dependencies to what is returned so that I could re-use these functions.

Reply:
I think it would make sense to only have a field called `'dependencies'`. The versions from the dependencies can be removed on line 322 before they are used to create the dep tree. In the future, we might want to do something with the versions there anyways.

Reply:
That makes sense. I'll make this change.

Reply:
@bnemanich 
See comments on how I implemented this.